pkg/ottl/compare.go

```diff
@@ -135,6 +136,48 @@ func (p *Parser[K]) compareFloat64(a float64, b any, op compareOp) bool {
 	}
 }
 
+func (p *Parser[K]) compareTime(a time.Time, b any, op compareOp) bool {
+	switch v := b.(type) {
+	case time.Time:
+		switch op {
+		case EQ:
+			if a.Equal(v) == true {
+				return true
+			}
+			return false
+		case NE:
+			if a.Equal(v) == false {
+				return true
+			}
+			return false
+		case LT:
+			if a.Before(v) == true {
+				return true
+			}
+			return false
+		case LTE:
+			if a.Before(v) == true || a.Equal(v) == true {
+				return true
+			}
+			return false
```

Comment:
For these we can return `a.Before(v) || a.Equal(v)`

Reply:
Done!